pr_agent/git_providers/bitbucket_server_provider.py

```diff
@@ -0,0 +1,353 @@
+
+        diff_files = []
+        original_file_content_str = ""
+        new_file_content_str = ""
+
+        changes = self.bitbucket_client.get_pull_requests_changes(self.workspace_slug, self.repo_slug, self.pr_num)
+        for change in changes:
+            file_path = change['path']['toString']
+            match change['type']:
+                case 'ADD':
+                    edit_type = EDIT_TYPE.ADDED
+                    new_file_content_str = self.get_file(file_path, head_sha)
+                    if isinstance(new_file_content_str, (bytes, bytearray)):
+                        new_file_content_str = new_file_content_str.decode("utf-8")
+                    original_file_content_str = ""
+                case 'DELETE':
+                    edit_type = EDIT_TYPE.DELETED
+                    new_file_content_str = ""
+                    original_file_content_str = self.get_file(file_path, base_sha)
+                    if isinstance(original_file_content_str, (bytes, bytearray)):
+                        original_file_content_str = original_file_content_str.decode("utf-8")
+                case 'RENAME':
+                    edit_type = EDIT_TYPE.RENAMED
+                case _:
+                    edit_type = EDIT_TYPE.MODIFIED
+                    original_file_content_str = self.get_file(file_path, base_sha)
+                    if isinstance(original_file_content_str, (bytes, bytearray)):
+                        original_file_content_str = original_file_content_str.decode("utf-8")
+                    new_file_content_str = self.get_file(file_path, head_sha)
+                    if isinstance(new_file_content_str, (bytes, bytearray)):
+                        new_file_content_str = new_file_content_str.decode("utf-8")
+
+            patch = load_large_diff(file_path, new_file_content_str, original_file_content_str)
+
+            diff_files.append(
+                FilePatchInfo(
+                    original_file_content_str,
+                    new_file_content_str,
+                    patch,
+                    file_path,
+                    edit_type=edit_type,
+                )
+            )
+
+        self.diff_files = diff_files
+        return diff_files
+
+    def publish_comment(self, pr_comment: str, is_temporary: bool = False):
+        if not is_temporary:
+            self.bitbucket_client.add_pull_request_comment(self.workspace_slug, self.repo_slug, self.pr_num, pr_comment)
+
+    def remove_initial_comment(self):
+        try:
+            for comment in self.temp_comments:
+                self.remove_comment(comment)
+        except ValueError as e:
+            get_logger().exception(f"Failed to remove temp comments, error: {e}")
+
+    def remove_comment(self, comment):
+        pass
+
+    # funtion to create_inline_comment
+    def create_inline_comment(self, body: str, relevant_file: str, relevant_line_in_file: str):
```

Comment:
**Suggestion:** Add type hints to the function parameters of `BitbucketServerProvider.create_inline_comment()`.
```suggestion
    def create_inline_comment(
      self, body: str, relevant_file: str, relevant_line_in_file: str
    ) -> dict:
```